Lib/test/test_weakref.py

```diff
@@ -391,6 +391,23 @@ class List(list): pass
         lyst = List()
         self.assertEqual(bool(weakref.proxy(lyst)), bool(lyst))
 
+    def test_proxy_iter(self):
+        # Test fails with a debug build of the interpreter
+        # (see bpo-38395).
+
+        obj = None
+
+        class MyObj:
+            def __iter__(self):
+                nonlocal obj
+                del obj
+                return NotImplemented
+
+        obj = MyObj()
+        p = weakref.proxy(obj)
+        with self.assertRaises(TypeError):
+            "blech" in p
```

Comment:
The relationship between "a in b" and `__iter__()` is nonobvious. Are you also able to reproduce the crash using something like:

```
# call indirectly __iter__()
for item in p: pass
```

Or maybe just add a comment to explain that in this case, "a in b" calls `__iter__()`.

Reply:
> Are you also able to reproduce the crash using something like

No, because that keeps a reference to the object, not triggering the problem. Will add a comment to explain that